conda_index/index/__init__.py

```diff
@@ -659,22 +645,45 @@ def index(
                 log.debug("found subdirs %s" % detected_subdirs)
                 self.subdirs = subdirs = sorted(detected_subdirs | {"noarch"})
             else:
-                self.subdirs = subdirs = sorted(
-                    set(self._subdirs)
-                )  # XXX noarch removed for testing | {"noarch"})
+                self.subdirs = subdirs = sorted(set(self._subdirs))
+                log.warn("Indexing %s does not include 'noarch'", subdirs)
 
             # Step 1. Lock local channel.
             with utils.try_acquire_locks(
                 [utils.get_lock(self.channel_root)], timeout=900
             ):
+                # begin non-stop "extract packages into cache";
+                # extract_subdir_to_cache manages subprocesses. Keeps cores busy
+                # during write/patch/update channeldata steps.
+                def extract_subdirs_to_cache():
+                    executor = ThreadPoolExecutor(max_workers=1)
+
+                    def extract_args():
+                        for subdir in self.subdirs:
+                            cache = self.cache_for_subdir(subdir)
+                            subdir_path = join(self.channel_root, subdir)
+                            yield (subdir, verbose, progress, subdir_path, cache)
+
+                    def extract_wrapper(args):
+                        cache = args[-1]
+                        with closing(cache.db):
```

Comment:
`db.close()` is necessary to prevent deadlocks. Must be an implicit transaction in our process' update-stat code that runs in a thread instead of in a subprocess?